ui/admin/app/routes/scopes/scope/sessions/index.js

```diff
@@ -4,5 +4,196 @@
  */
 
 import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
+import { notifySuccess, notifyError } from 'core/decorators/notify';
 
-export default class ScopesScopeSessionsIndexRoute extends Route {}
+export default class ScopesScopeSessionsIndexRoute extends Route {
+  // =services
+
+  @service store;
+  @service resourceFilterStore;
+
+  // =attributes
+
+  queryParams = {
+    search: {
+      refreshModel: true,
+      replace: true,
+    },
+    users: {
+      refreshModel: true,
+      replace: true,
+    },
+    targets: {
+      refreshModel: true,
+      replace: true,
+    },
+    status: {
+      refreshModel: true,
+      replace: true,
+    },
+    page: {
+      refreshModel: true,
+    },
+    pageSize: {
+      refreshModel: true,
+    },
+  };
+
+  allUsers;
+  allSessions;
+
+  /**
+   * Loads all sessions under the current scope and encapsulates them into
+   * an array of objects containing their associated users and targets.
+   * @return {Promise{[{sessions: [SessionModel], allSessions: [SessionModel], allUsers: [UserModel], allTargets: [TargetModel], totalItems: number}]}}
+   */
+  async model({ search, users, targets, status, page, pageSize }) {
+    const { id: scope_id } = this.modelFor('scopes.scope');
+    const filters = {
+      status: [],
+      user_id: [],
+      target_id: [],
+    };
+    users.forEach((user) => {
+      filters.user_id.push({ equals: user });
+    });
+    targets.forEach((target) => {
+      filters.target_id.push({ equals: target });
+    });
+    status.forEach((item) => {
+      filters.status.push({ equals: item });
+    });
+
+    const queryOptions = {
+      scope_id,
+      recursive: true,
+      query: { search, filters },
+      page,
+      pageSize,
+      force_refresh: true,
+    };
+
+    const sessions = await this.store.query('session', queryOptions);
+    const totalItems = sessions.meta?.totalItems;
+
+    sessions.forEach(async (session) => {
```

Comment:
perhaps we should replace `forEach` due to it [not awaiting](https://gist.github.com/joeytwiddle/37d2085425c049629b80956d3c618971) for promises to resolve.

Reply:
I actually removed the whole thing based on feedback from another PR comment.